### ISSUE TITLE
SetUpFixture dows not work with static correctly

### ISSUE THREAD

**Issue by NUnitTester** [nunit/nunit#3363]
OneTimeSetUp can be static and works. The following code does what expected.
```
[SetUpFixture]
public class SetUpAllClasses
{
  [OneTimeSetUp]
  public static void OneTimeSetUp()
  {
  ...
  }

  [OneTimeTearDown]
  public static void OneTimeTearDown()
  {
  ...
  }
}
```

But if we set the class also to static, what is purposed because it does not contain any non-static methods, OneTimeSetUp does not works any more. Thats an error.

```
[SetUpFixture]
public static class SetUpAllClasses { ... }
```

**Comment by mikkelbu** [comment]
@NUnitTester Can you provide some more information as the example below works on my machine.
* which version of NUnit
* how do you run the tests and which version of the this tool
* .NET framework or .NET core (and version)

```csharp
namespace NUnitTest
{
    [SetUpFixture]
    public static class SetUpAllClasses
    {
        public static bool HasRun;

        [OneTimeSetUp]
        public static void OneTimeSetUp()
        {
            HasRun = true;
        }

        [OneTimeTearDown]
        public static void OneTimeTearDown()
        {
            Assert.True(SetUpAllClasses.HasRun);
        }
    }

    public class Tests
    {
        [Test]
        public void Test1()
        {
            Assert.True(SetUpAllClasses.HasRun);
        }
    }
}
```

**Comment by NUnitTester** [comment]
Ok, your example doesn't work in my environment, too.

Error Message:
OneTimeSetUp: NUnitTest.SetUpAllClasses is an abstract class
  Exception doesn't have a stacktrace

I'm using Resharper 2019.2 - might be a problem?

.NET Framework 4.6.1  
compiled for x86
NUnit 3.9

**Comment by NUnitTester** [comment]
Updated now to NUnit 3.12
and also found NUnit3TestAdapter updated to  3.15

Now it works! One of the Update should be important one.
Sorry for that.

**Comment by mikkelbu** [comment]
@NUnitTester You are right. The update to NUnit 3.12 solved the problem, as this issue was reported in #3166 and solved in #3217 which was part of 3.12.
